File: FairPlay_Streaming_Server_SDK_5.1/Development/Key_Server_Module/Swift/Sources/src/extension/structures/extension_structures.swift
//
// Copyright © 2023-2024 Apple Inc. All rights reserved.
//

public struct SDKExtension {}

public struct FPSOperationExtension {}

public struct AssetInfoExtension {
  var contentType: extension_constants.ContentType = .unknown
}

public struct ServerCtxExtension {
  var contentType: extension_constants.ContentType = .unknown
}

public struct FPSResultsExtension {}

public struct FPSResultExtension {}

public struct SPCDataExtension {}

public struct CKCDataExtension {
  var requiredSecurityLevel: extension_constants.FPSSecurityLevel = .main
}

public struct SPCContainerExtension {}

public struct ClientFeaturesExtension {}

public struct KeyDurationExtension {}

File: FairPlay_Streaming_Server_SDK_5.1/Development/Key_Server_Module/Rust/src/extension/structures/extension_structures.rs
//
// Copyright © 2023-2024 Apple Inc. All rights reserved.
//

use crate::extension::extension_constants::ContentType;
use crate::extension::extension_constants::FPSSecurityLevel;
use std::fmt::Debug;

#[derive(Debug, Default, Clone)]
pub struct SDKExtension {}

#[derive(Debug, Default, Clone)]
pub struct FPSOperationExtension {}

#[derive(Debug, Default, Clone)]
pub struct AssetInfoExtension {
    pub contentType: ContentType,
}

#[derive(Debug, Default, Clone)]
pub struct ServerCtxExtension {
    pub contentType: ContentType,
}

#[derive(Debug, Default, Clone)]
pub struct FPSResultsExtension {}

#[derive(Debug, Default, Clone)]
pub struct FPSResultExtension {}

#[derive(Debug, Default, Clone)]
pub struct SPCDataExtension {}

#[derive(Debug, Default, Clone)]
pub struct CKCDataExtension {
    pub requiredSecurityLevel: FPSSecurityLevel,
}

#[derive(Debug, Default, Clone)]
pub struct SPCContainerExtension {}

#[derive(Debug, Default, Clone)]
pub struct ClientFeaturesExtension {}

#[derive(Debug, Default, Clone)]
pub struct KeyDurationExtension {}
